src/main/java/com/pi4j/driver/video/st7789/St7789Driver.java

```diff
@@ -0,0 +1,327 @@
+package com.pi4j.driver.video.st7789;
+
+import java.awt.image.BufferedImage;
+import java.awt.image.DataBuffer;
+import java.awt.image.DataBufferByte;
+import java.awt.image.DataBufferInt;
+import java.io.IOException;
+
+import com.pi4j.io.gpio.digital.DigitalOutput;
+import com.pi4j.io.spi.Spi;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.pi4j.driver.interfaces.Video;
+import com.pi4j.driver.LedColor;
+
+public class St7789Driver implements Video {
+
+    private static Logger log = LoggerFactory.getLogger(St7789Driver.class);
+
+    private final int BITS_PER_PIXEL = 16;
+    private final int OFFSET = 80;
+    private final int WIDTH = 240;
+    private final int HEIGHT = 240;
+
+    private final byte[] image = new byte[WIDTH * HEIGHT * BITS_PER_PIXEL / 8];
+
+    private static final int SWRESET = 0x01;
+    private static final int SLPOUT = 0x11;
+    private static final int NORON = 0x13;
+    private static final int INVON = 0x21;
+    private static final int DISPON = 0x29;
+    private static final int CASET = 0x2A;
+    private static final int RASET = 0x2B;
+    private static final int RAMWR = 0x2C;
+    private static final int MADCTL = 0x36;
+    private static final int COLMOD = 0x3A;
+
+    private Spi spi;
+    private DigitalOutput dc;
+
+    public St7789Driver(Spi spi, DigitalOutput dc) {
+
+        this.spi = spi;
+        this.dc = dc;
+
+        try {
+            init();
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private void init() throws Exception {
+
+        command(SWRESET);
+
+        command(SLPOUT);
+
+        command(COLMOD);
+        data(0x55);
+
+        command(MADCTL);
+        data(0x08);
+
+        command(CASET); // Column addr set
+        byte[] cols = new byte[4];
+        cols[0] = 0x00;
+        cols[1] = 0x00;
+        cols[2] = (byte) (WIDTH >> 8);
+        cols[3] = (byte) (WIDTH & 0xff);
+        data(cols);
+
+        command(RASET); // Row addr set
+        byte[] rows = new byte[4];
+        rows[0] = 0x00;
+        rows[1] = 0x50;
+        rows[2] = (byte) ((OFFSET + HEIGHT) >> 8);
+        rows[3] = (byte) ((OFFSET + HEIGHT) & 0xff);
+        data(rows);
+
+        command(INVON);
+
+        command(NORON);
+
+        command(DISPON);
+
+        command(MADCTL);
+        data(0xC0);
+
+    }
+
+    private void command(int x) throws com.pi4j.io.exception.IOException, IOException {
+
+        if (x < 0 || x > 0xff) {
+            throw new IllegalArgumentException("ST7789 bad command value " + x);
+        }
+
+        log.trace("Command: " + x);
```

Comment:
for logging can we use the comma separated version instead of concatenating?

    log.trace("Command: {}", x);